src/gattlib.cpp

```diff
@@ -84,6 +84,12 @@ GATTResponse::on_response(boost::python::object data) {
 void
 GATTResponse::notify(uint8_t status) {
     _status = status;
+    _complete = true;
+    if (!status) {
+      on_response_complete();
+    } else {
+      on_response_failed(status);
+    }
```

Comment:
I have a doubt about these. Here, the method only sets the `_event` in order to wake up pending threads, but the thread that does the notify will not go inside Python land. I'm not sure if this change would present some problem in that context. @jsgh what do you think?

Reply:
This notification is only made if the caller has already acquired the GIL, and allows the python code to override those functions to get immediate notification of completion. (It can then do some work for that immediately on the IOThread itself, though anything substantial likely requires it to use a multiprocessing.Queue or similar to coordinate back to the main thread. Or a worker of its own.)

Or the python can just ignore them (in which case those two calls above are no-ops) and poll for completion on the main thread. (Though it should use the new  complete() method rather than polling received() as in the current example code - that has problems distinguishing between actual completion, partial completion, completion with no results, and error.)

Or it can call wait() which I have now exposed for a non-polling solution.

It's a matter of taste and what fits with the rest of the client's implementation.

Reply:
Ok, thanks for explaining it. :)